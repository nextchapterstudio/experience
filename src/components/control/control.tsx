import React, { FC, ElementType } from 'react'
import classnames from 'classnames'
import { BaseElement } from '../base-element'

export interface ControlProps {
  hasIconsLeft?: boolean
  hasIconsRight?: boolean
  isExpanded?: boolean
  isGrouped?: boolean
  as?: ElementType
  children?: React.ReactNode
  className?: string
}
export const Control: FC<ControlProps> = ({
  as,
  className,
  children,
  hasIconsLeft,
  hasIconsRight,
  isExpanded,
  isGrouped,
}) => {
  const classes = {
    'has-icons-left': hasIconsLeft,
    'has-icons-right': hasIconsRight,
    'is-expanded': isExpanded,
    'is-grouped': isGrouped,
  }
  return (
    <BaseElement as={as} className={classnames('control', className, classes)}>
      {children}
    </BaseElement>
  )
}

Control.defaultProps = {
  as: 'div',
}
