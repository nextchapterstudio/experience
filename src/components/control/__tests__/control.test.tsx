import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { Control } from '../control'

describe('Container component', () => {
  afterEach(cleanup)
  it('should render', () => {
    const { container } = render(<Control>Some Text</Control>)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('control')
  })
  it('should render with extra classnames', () => {
    const { container } = render(<Control className="contained">Contained Text</Control>)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('control')
    expect(container.firstChild).toHaveClass('contained')
  })
  it('should render with isSmall', () => {
    const { container } = render(<Control isExpanded>Expanded Control</Control>)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('control')
    expect(container.firstChild).toHaveClass('is-expanded')
  })
  it('should render with isGrouped', () => {
    const { container } = render(<Control isGrouped>Grouped Control</Control>)

    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('control')
    expect(container.firstChild).toHaveClass('is-grouped')
  })
})
