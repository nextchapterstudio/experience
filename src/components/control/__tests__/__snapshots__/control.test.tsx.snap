// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component should render 1`] = `
<div
  class="control"
>
  Some Text
</div>
`;

exports[`Container component should render with extra classnames 1`] = `
<div
  class="control contained"
>
  Contained Text
</div>
`;

exports[`Container component should render with isGrouped 1`] = `
<div
  class="control is-grouped"
>
  Grouped Control
</div>
`;

exports[`Container component should render with isSmall 1`] = `
<div
  class="control is-expanded"
>
  Expanded Control
</div>
`;
